static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -4,127 +4,153 @@ import capitalize from 'lodash/capitalize';
 import pick from 'lodash/pick';
 
 import {t, tct} from 'sentry/locale';
-import {ExternalActorMapping, Integration, Organization} from 'sentry/types';
+import {ExternalActorMapping, Integration} from 'sentry/types';
+import {getExternalActorEndpointDetails} from 'sentry/utils/integrationUtil';
 import {FieldFromConfig} from 'sentry/views/settings/components/forms';
 import Form from 'sentry/views/settings/components/forms/form';
+import FormModel from 'sentry/views/settings/components/forms/model';
 import {Field} from 'sentry/views/settings/components/forms/type';
 
-type Props = Pick<Form['props'], 'onSubmitSuccess' | 'onCancel'> &
-  Partial<Pick<Form['props'], 'onSubmit'>> & {
-    organization: Organization;
-    integration: Integration;
-    mapping?: ExternalActorMapping;
-    type: 'user' | 'team';
-    baseEndpoint?: string;
-    sentryNamesMapper: (v: any) => {id: string; name: string}[];
-    url: string;
-    onResults?: (data: any) => void;
-  };
+type Props = Pick<Form['props'], 'onCancel' | 'onSubmitSuccess' | 'onSubmitError'> & {
+  integration: Integration;
+  mapping?: ExternalActorMapping;
+  type: 'user' | 'team';
+  getBaseFormEndpoint: (mapping?: ExternalActorMapping) => string;
```

Comment:
This method is passed in because the endpoint we're submitting to is actually determined by the info in the form, at least for team mappings. E.g., If updating externalName `@frontend` to point to sentryTeam `#frontend` instead of `#triage`, we'd submit to `/teams/frontend/external-teams/` or something along those lines. This complicates things and prevents us from passing in a static form endpoint.